azuredevops/internal/service/serviceendpoint/resource_serviceendpoint_gcp.go

```diff
@@ -0,0 +1,83 @@
+package serviceendpoint
+
+import (
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v6/serviceendpoint"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/converter"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/tfhelper"
+)
+
+// ResourceServiceEndpointGcp schema and implementation for gcp service endpoint resource
+func ResourceServiceEndpointGcp() *schema.Resource {
+	r := genBaseServiceEndpointResource(flattenServiceEndpointGcp, expandServiceEndpointGcp)
+	r.Schema["client_email"] = &schema.Schema{
+		Type:        schema.TypeString,
+		Required:    false,
+		DefaultFunc: schema.EnvDefaultFunc("AZDO_GCP_SERVICE_CONNECTION_CLIENT_EMAIL", nil),
+		Description: "The client email field in the JSON key file for creating the JSON Web Token.",
+	}
+	r.Schema["private_key"] = &schema.Schema{
+		Type:             schema.TypeString,
+		Required:         true,
+		DefaultFunc:      schema.EnvDefaultFunc("AZDO_GCP_SERVICE_CONNECTION_PRIVATE_KEY", nil),
+		Description:      "Private Key for connecting to the endpoint.",
+		Sensitive:        true,
+		DiffSuppressFunc: tfhelper.DiffFuncSuppressSecretChanged,
+	}
+	saSecretHashKey, saSecretHashSchema := tfhelper.GenerateSecreteMemoSchema("private_key")
+	r.Schema[saSecretHashKey] = saSecretHashSchema
+	r.Schema["token_uri"] = &schema.Schema{
+		Type:        schema.TypeString,
+		Optional:    false,
+		DefaultFunc: schema.EnvDefaultFunc("AZDO_GCP_SERVICE_CONNECTION_TOKEN_URI", nil),
+		Description: "The token uri field in the JSON key file for creating the JSON Web Token.",
+		Sensitive:   false,
+	}
+	r.Schema["scope"] = &schema.Schema{
+		Type:        schema.TypeString,
+		Optional:    true,
+		DefaultFunc: schema.EnvDefaultFunc("AZDO_GCP_SERVICE_CONNECTION_SCOPE", nil),
+		Description: "Scope to be provided",
+	}
+	r.Schema["gcp_project_id"] = &schema.Schema{
+		Type:        schema.TypeString,
+		Optional:    false,
```

Comment:
if this property is required,  use `Required:    true,` and  if optional use `Optional:    true,`